[Dominic Siew Zhen Yu] iP

Domszy opened the pull request:
### DukeMan 

> "You will meet many defeats in life, but never let yourself be defeated!" (source: [Top 100 most famous quotes](https://blog.hubspot.com/sales/famous-quotes)) 

Manage your tasks with DukeMan, an all-in-one task management application. 🙂
- Categorise your tasks as deadlines, todo or events
- ~~FAST~~ SUPER FAST and easy to use 
- Great user interface

All you need to do to use DukeMan 👀
1. Download DukeMan.
2. Run the program on an IDE.
3. Use the todo, deadline and event commands to insert tasks into DukeMan.
4. Use the "remove" command to remove tasks, and the "done" command to mark the task as complete. 
5. Use "list" to have a look at your tasks.
6. Use "bye" command to quite DukeMan. 

**DOWNLOAD NOW!** 

- [ ] Download DukeMan to enhance your efficiency! 
- [ ] What are you waiting for? 


If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Domszy committed:
Update from AY2021 Semester 1 repository

Domszy committed:
Level-1

Domszy committed:
merged with the previous repository

Domszy committed:
Level-2

Domszy committed:
Level-3

Domszy committed:
Level-4

Domszy committed:
Level-5

Domszy committed:
A-TextUiTesting

Domszy committed:
Level-6

Domszy committed:
Level-7

Domszy committed:
Level-8

Domszy committed:
Merge branch 'branch-Level-7' into branch-Level-8

Domszy committed:
A-MoreOOP

Domszy committed:
A-Packages

Domszy committed:
A-JUnit

Domszy committed:
A-Jar

Domszy committed:
A-JavaDoc

Domszy committed:
A-CodingStandard

Domszy committed:
Level-9

Domszy committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

Domszy committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

Domszy committed:
Update to the "find" function in DukeMan

Domszy committed:
A-Gradle

Domszy committed:
A-CheckStyle

nataniayp commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,43 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task{
```
You might want to change the class name to `Deadline` instead since plural forms are usually for a collection of data such as an array of deadlines. 😄

Javiier-pzk replied:
You may want to change your class name to singular form (`deadline`) as plural forms are usually for collection of items.

nataniayp commented on src/main/java/duke/DukeMan.java:
```diff
@@ -0,0 +1,109 @@
+            String command = parser.getCommand();
+
+            switch(command) {
+                case "todo":
```
the indentation for `case` should be the same as `switch`

nataniayp commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+     *
+     * @return a string representation of a Event object
+     */
+    public String printName() {
```
instead of `printName`, you might want to override the `toString` method instead since your method technically does not print the name

nataniayp commented on src/main/java/duke/taskList.java:
```diff
@@ -0,0 +1,267 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class taskList {
```
class name should've been capitalised instead -- so `TaskList` instead of `taskList` ?

Javiier-pzk replied:
As part of convention, your class name should have been capitalised. It should be `TaskList` instead of `taskList`

nataniayp reviewed (commented):
overall it's fine! just a bit of naming conventions but otherwise looks good 👍

Javiier-pzk commented on src/main/java/duke/DukeMan.java:
```diff
@@ -0,0 +1,109 @@
+                    String todoName = parser.getTaskName();
+                    tasks.addTodo(todoName, true);
+                    break;
+                case "deadline":
```
`case` statement should have the same indentation level as `switch`

Javiier-pzk commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     * @param input the given input by the user
+     */
+
+    public void parsing(String input) {
```
Perhaps a better name for this method will be `parse`? I think we generally should avoid adding '-ing' behind our methods and class names.

Javiier-pzk commented on src/main/java/duke/DukeMan.java:
```diff
@@ -0,0 +1,109 @@
+ * @author Dominic Siew Zhen Yu
+ */
+
+public class DukeMan {
```
Perhaps you can name your class name in a way that explains what this class is supposed to do, as per the naming guidelines

Javiier-pzk commented on src/main/java/duke/taskList.java:
```diff
@@ -0,0 +1,267 @@
+                boolean isCompleted = (seperate2[0].equals("[✓]"))? true: false;
+                String eventInfo = seperate2[1];
+
+                switch(taskTypeText) {
```
Same indentation problem with `switch` and `case`

Javiier-pzk commented on src/main/java/duke/taskList.java:
```diff
@@ -0,0 +1,267 @@
+
+            while ((currentLine = reader.readLine()) != null) {
+                String[] seperate1 = currentLine.toString().split(" ", 2);
+                String taskTypeText = seperate1[0];
```
Perhaps a better name would be `taskString` ?

Javiier-pzk commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,137 @@
+//
+///**
+// * Duke is a chatbot that takes in tasks, which includes todos, deadlines and events, and compiles them
+// * in a list. Tasks can be removed from the list and be marked as completed.
```
Just a friendly reminder that there should not be large chunks of commented code present as mentioned in the iP grading rubrics. You should delete them if not needed or transfer them somewhere else 😄

Javiier-pzk reviewed (commented):
Hi Dominic! Your code looks good overall but there are just a few naming and coding style issues that you should fix. Cheers!

Domszy committed:
Level-10

Domszy committed:
no message

Domszy committed:
Merge branch 'branch-A-Gradle' into branch-Level-10

Domszy committed:
A-Assertions

Domszy committed:
Amended for greater readability.

Such amendments are necessary so that other programmers who come across my forked repository will be able to understand the program.

Removed duplicated code, adding default branches for if,else statements, removed the Duke class which was redundant, simplified several expressions.

Makes the code easier to understand and read, especially for programmers who are not aware of my program.

Domszy committed:
Addition of the scheduling function

Addition of the TimeComparator class and the Schedule class. Additional
method to the taskList

Provide the users with a compiled list of events and deadline for the
day they have inputted

Created the schedule class to manage the processing of the data. The
Timecomparator class is used to sort the ArrayList of tasks.

Ensures greater abstraction from the TaskList class and the DukeMan
class.

Domszy committed:
Merge pull request #3 from Domszy/branch-A-Assertions

A-Assertions

Domszy committed:
Merge branch 'master' into branch-A-CodeQuality

Domszy committed:
Merge pull request #2 from Domszy/branch-A-CodeQuality

A-CodeQuality

Domszy committed:
Merge branch 'master' into branch-BCD-Extension

Domszy committed:
Merge pull request #1 from Domszy/branch-BCD-Extension

Addition of the scheduling function

Domszy committed:
branch-JustToCheck

Domszy committed:
A-CI

Domszy committed:
A-BetterGui

Domszy committed:
A-BetterGui

Domszy committed:
Set theme jekyll-theme-merlot

Domszy committed:
A-UserGuide

Domszy committed:
Add files via upload

Domszy committed:
Update README.md

Domszy committed:
Add files via upload

Domszy committed:
Update README.md

Domszy committed:
Set theme jekyll-theme-slate

Domszy committed:
Set theme jekyll-theme-slate

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
Delete Ui.png

Domszy committed:
Add files via upload

Domszy committed:
A-UserGuide

Domszy committed:
Merge commit '6981a3bb8c9bd3715f8ac8ba5bba0aec47a9fdea' into branch-JustToCheck

Domszy committed:
A-UserGuide

Domszy committed:
Update README.md

Domszy committed:
Add files via upload

Domszy committed:
Set theme jekyll-theme-slate

Domszy committed:
Fixed issues when pushing to repository

Domszy committed:
test tp user guide in iP

Domszy committed:
test tP's user guide

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
Set theme jekyll-theme-cayman

Domszy committed:
Update README.md

Domszy committed:
Update README.md

Domszy committed:
no message

Domszy committed:
Merge commit 'd6bdc6f3d02334bc5b96ca48ec75a10c78365ca3'

* commit 'd6bdc6f3d02334bc5b96ca48ec75a10c78365ca3':
  Update README.md
  Update README.md
  Set theme jekyll-theme-cayman
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md

# Conflicts:
#	docs/README.md

Domszy committed:
Update README.md
